src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
```

Comment:
This can be more readable by improving like, e.g:

`children.forEach((ch) => {`
    `const mother = people.find((person) => person.name === ch.mother);`
    `const sum = ch.born - mother.born;`
    `return sum;`
 ` })`
  `.reduce((total, filteredSum) => total + filteredSum, 0);`